src/cosmicqc/cli.py

```diff
@@ -0,0 +1,111 @@
+"""
+Setup coSMicQC CLI through python-fire
+"""
+
+import inspect
+import sys
+import types
+from typing import Optional
+
+import fire
+from fire import helptext, inspectutils, value_types
+from fire.core import Display, FireError, _DictAsString, _OneLineResult
+from fire.trace import FireTrace
+
+from . import analyze
+
+
+# referenced from https://github.com/google/python-fire/pull/446
+# to be removed after python-fire merges changes (uncertain of timeline)
+def HasCustomRepr(component: object) -> bool:
+    """Reproduces above HasCustomStr function to determine if component has a
+    custom __repr__ method.
+
+    ...
+
+    Args:
+      component: The object to check for a custom __repr__ method.
+    Returns:
+      Whether `component` has a custom __repr__ method.
+    """
+    if hasattr(component, "__repr__"):
+        class_attrs = inspectutils.GetClassAttrsDict(type(component)) or {}
+        repr_attr = class_attrs.get("__repr__")
+        if repr_attr and repr_attr.defining_class is not object:
+            return True
+    return False
+
+
+# referenced with modifications from https://github.com/google/python-fire/pull/446
+# to be removed after python-fire merges changes (uncertain of timeline)
+# ruff: noqa: C901
```

Comment:
Is this a linting thing like you mentioned before?

Reply:
Might recommend adding a short comment mentioning what it is for the future reader to not be confused.

Reply:
Thanks for checking on this - yes this is an ignore rule for ruff for [C901](https://docs.astral.sh/ruff/rules/complex-structure/). In this case I didn't do much modification of the code as I wasn't sure how it might change in the future. I'll document this better for the time being.